hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/migration/InitializeEntityBalanceMigration.java

```diff
@@ -87,4 +103,17 @@ protected void doMigrate() {
         int count = jdbcOperations.update(INITIALIZE_ENTITY_BALANCE_SQL);
         log.info("Initialized {} entities balance in {}", count, stopwatch);
     }
+
+    @TransactionalEventListener(classes = InitializeEntityBalanceEvent.class)
+    public void reRunMigration() {
+        log.info("Re-running the InitializeEntityBalanceMigration on InitializeBalanceEvent");
+        doMigrate();
+    }
+
+    @Override
+    public void onEnd(AccountBalanceFile streamFile) throws ImporterException {
+        if (accountBalanceFileRepository.count() == 0) {
```

Comment:
A count query in postgresql is extremely slow and should never be used. You should be able to use the existing repository query to find the earliest row.
```suggestion
        if (accountBalanceFileRepository.findNextInRange(0, Long.MAX_VALUE).isEmpty()) {
```